wikimedia/wikipedia-ios
# Populate notification cells with live data

**Phabricator:** https://phabricator.wikimedia.org/T288662

### Notes
This PR populates the notification cells created in https://github.com/wikimedia/wikipedia-ios/pull/4041 with data from the `RemoteNotification` managed object. 

This is only meant to fill in the managed object -> view model -> cell transformations so that we can view our cells with live data. We aren't handling any linking yet - tap and swipe actions will need to be handled in a later PR. Even without that this PR is rather large! I'm happy to break it up further if needed. If not, the commit history is pretty clean and probably the best way to review this.

Note this PR is based off of https://github.com/wikimedia/wikipedia-ios/pull/4041, so that will need to be merged first.

### Test Steps
1. Run on the main/production scheme with an account that has lots of notifications. It's helpful if you're able to find an account with notifications on Commons, Wikidata, and multiple languages. Be sure to add those other languages as an app language in Settings before going to Notification Center. I can share my testing credentials if needed.
2. Browse through notification center via the Explore bell icon. Make sure nothing looks obviously off by comparing with the Figma notification types. There are a lot of types, so we will likely need to lean on QA for a more thorough test of this logic.
